utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -0,0 +1,154 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! `Structopt`-ready struct for `try-runtime`.
+
+use parity_scale_codec::{Decode, Encode};
+use std::{fmt::Debug, str::FromStr};
+use sc_service::Configuration;
+use sc_cli::{CliConfiguration, ExecutionStrategy};
+use sc_executor::{WasmExecutionMethod, NativeExecutor};
+use sp_state_machine::StateMachine;
+use sc_service::NativeExecutionDispatch;
+use sp_runtime::traits::{Block as BlockT, NumberFor};
+use sp_core::storage::{StorageData, StorageKey, well_known_keys};
+use frame_try_runtime::Target;
+
+/// Various commands to try-out (similar to `TryFrom`, `TryInto`) the new runtime, over configurable
+/// states.
+///
+/// For now this only assumes running the runtime-upgrade hooks.
+#[derive(Debug, structopt::StructOpt)]
+pub struct TryRuntimeCmd {
+	/// The shared parameters
+	#[allow(missing_docs)]
+	#[structopt(flatten)]
+	pub shared_params: sc_cli::SharedParams,
+
+	/// The target pallet to run the migration against.
+	#[structopt(short, long, default_value = "All")]
+	pub target: Target,
+
+	/// The state to use to run the migration. Should be a valid FILE or HTTP URI.
+	#[structopt(short, long, default_value = "http://localhost:9933")]
+	pub state: State,
+}
+
+/// The state to use for a migration dry-run.
+#[derive(Debug)]
+pub enum State {
+	/// A snapshot. Inner value is file path.
+	Snap(String),
+
+	/// A live chain. Inner value is the HTTP uri.
+	Live(String),
+}
+
+impl FromStr for State {
+	type Err = &'static str;
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		match s.get(..7) {
+			// could use Url crate as well, but lets keep it simple for now.
+			Some("http://") => Ok(State::Live(s.to_string())),
```

Comment:
mmh, this means https is not supported, though :cry:

Reply:
I don't we even expose http for RPC? afaik it is only ws (safe) and http (yolo).